### ISSUE TITLE
Install the Rust toolchain and target for wasm

### ISSUE THREAD

**Issue by yutannihilation** [r-wasm/rwasm#20]
I noticed R-universe's wasm build installs `rustc` and `cargo` if there's the `SystemRequirements`.

```
> rwasm::build('./savvyExamples_0.0.1.tar.gz')
 
> Will download 1 package with unknown size.
> Will install 2 system packages:
+ cargo  - savvyExamples-deps
+ rustc  - savvyExamples-deps
```
https://github.com/r-universe/yutannihilation/actions/runs/7848104729/job/21418576850#step:5:324

However, the build fails because the `cargo` is not the supposed one.

```
...snip...

    cargo +nightly build --lib --release --manifest-path=./rust/Cargo.toml --target-dir ./rust/target --target wasm32-unknown-emscripten -Zbuild-std=panic_abort,std; \
  fi
error: no such command: `+nightly`

	Cargo does not handle `+toolchain` directives.
	Did you mean to invoke `cargo` through `rustup` instead?
make: *** [Makevars:18: rust/target/wasm32-unknown-emscripten/release/libsimple_savvy.a] Error 101
```

I'm not sure where this should be handled. Options that I came up with are

* Install Rust into the base Docker image. But, this turned out the size is not very lightweight (cf. https://github.com/r-wasm/actions/pull/13#issuecomment-1920131274)
* R-universe or r-wasm might be able to provide a PPA to override the `cargo` and `rustc` DEB packages, and add the repository to the Docker image.
* pak package might be able to handle this, treating the Rust system requirements specially. This is probably not very clean. But, as pak (and pkgdepends) doesn't provide option to customize the internal processes, we might need to ask pak for some help like this.
* rwasm package might be able to install Rust toolchain on the fly. With pkgdepends, it's easy to detect if the package requires Rust like this:

``` r
deps <- pkgdepends::new_pkg_installation_proposal(".", config = list(sysreqs = TRUE, sysreqs_lookup_system = FALSE))
deps$solve()
any(vapply(deps$get_sysreqs(), \(x) "cargo" %in% names(x), logical(1L)))
```

**Comment by georgestagg** [comment]
Thanks for investigating all the available options for us, I appreciate it.

Okay, so after re-reading the thread in [r-wasm/actions#13](https://github.com/r-wasm/actions/pull/13) and this one I think updating the base Docker image is the best way to go. It's not ideal with the size increase, but I think once compressed it's not so bad (details below) and it means it won't have to be constantly reinstalled in GHA runs for rust R packages.

Let's install the required rust and `cargo` in the [r-wasm/webr](https://github.com/r-wasm/webr) base Dockerfile. Since you have this working already, would you mind opening a PR there adding the required Docker `RUN` commands?

<details>

```
$ docker manifest inspect ghcr.io/r-wasm/webr | grep size                   
		"size": 3558,
			"size": 30438977,
			"size": 645927155,
			"size": 497774382,
			"size": 32,
```

3558 + 30438977 + 645927155 + 497774382 + 32 = 1.17GB

```
$ docker manifest inspect ghcr.io/yutannihilation/webr-rust:main | grep size
		"size": 4960,
			"size": 30439111,
			"size": 629569132,
			"size": 509504104,
			"size": 32,
			"size": 209537935,
```

4960 + 30439111 + 629569132 + 509504104 + 32 + 209537935 = 1.38GB
</details>

**Comment by yutannihilation** [comment]
Wow, thank you! Okay, let me try a pull request. Before that I think, I need a bit more struggle:

* The last time I tried, the CI fails with `cargo: command not found` error (cf. https://github.com/yutannihilation/savvy/issues/84). Probably I needed to add `ENV` steps to the last squashing step, but I should confirm.
    * A related apology: After I found the CI error, I reverted the Dockerfile, so the image you inspected is not the same one as I reported on https://github.com/r-wasm/actions/pull/13#issuecomment-1920131274, sorry. I'll report the result of `docker manifest inspect` later. I guess probably it won't change much, though.
* Reading [R-universe's Docker image](https://github.com/r-universe-org/build-wasm/blob/51bbee600c268dd272ad6b62446079ebe043d453/Dockerfile#L15-L25), I learned the trick of installing a fake package using `equivs`. It might be better to do this for `rustc` and `cargo`; it will not only reduce the time of `rwasm::build()` a bit but also prevent troubles related to the multiple Rust installations. This is optional, and I can remove if you don't like this way.
